### ISSUE TITLE
[webview_flutter]java.lang.IllegalStateException: Could not find identifier for WebView

### ISSUE THREAD

**Issue by NamHofstadter** [flutter/flutter#119899]
```
webview_flutter: ^4.0.2
```
```
Fatal Exception: java.lang.IllegalStateException: Could not find identifier for WebView.
       at io.flutter.plugins.webviewflutter.WebViewClientFlutterApiImpl.onPageStarted(WebViewClientFlutterApiImpl.java:82)
       at io.flutter.plugins.webviewflutter.WebViewClientHostApiImpl$WebViewClientImpl.onPageStarted(WebViewClientHostApiImpl.java:49)
       at eC0.b(chromium-TrichromeWebViewGoogle.aab-stable-535907933:15)
       at ka.handleMessage(chromium-TrichromeWebViewGoogle.aab-stable-535907933:538)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:238)
       at android.os.Looper.loop(Looper.java:349)
       at android.app.ActivityThread.main(ActivityThread.java:8241)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1034)
```
same crash occurs in 
```
WebViewClientFlutterApiImpl.requestLoading
WebViewClientFlutterApiImpl.onProgressChanged
WebViewClientFlutterApiImpl.onReceivedRequestError
```

**Comment by NamHofstadter** [comment]
@exaby73

**Comment by huycozy** [comment]
Hi @NamHofstadter 
Which platform does the issue occur? Does it happen when running [plugin example](https://github.com/flutter/plugins/tree/main/packages/webview_flutter/webview_flutter/example)? 

If still so, please provide complete output of `flutter run -v` so that we can investigate and verify the issue. Thank you!

**Comment by NamHofstadter** [comment]
It occurs in Android device, I can't reproduce, but there is many crashes in our production app, seems caused by weak reference

**Comment by NamHofstadter** [comment]
@huycozy help me, please

**Comment by huycozy** [comment]
@NamHofstadter Can you please share a snippet code of webview implementation? And does the issue happen when running on `release` mode? 

Also, please provide the list of affected device info and the percentage occurrence of the crash. If there are remaining logs, please also share them so that it may be helpful for us to investigate the issue. Thanks!

**Comment by rounce** [comment]
same issue.
Android 11: 58%
Android 7: 17%
Android 10: 17%
Android 9: 8%

samsung 25%
Amlogic 17%
OnePlus 17%
TECNO MOBILE LIMITED 8%
Other: 33%

appeared after update 3.0.4 -> 4.0.1

```dart
    super.initState();
    late final PlatformWebViewControllerCreationParams params;
    if (WebViewPlatform.instance is WebKitWebViewPlatform) {
      params = WebKitWebViewControllerCreationParams(
        allowsInlineMediaPlayback: true,
        mediaTypesRequiringUserAction: const <PlaybackMediaTypes>{},
      );
    } else {
      params = const PlatformWebViewControllerCreationParams();
    }
    _controller = WebViewController.fromPlatformCreationParams(params);
    _controller
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..setBackgroundColor(const Color(0x00000000))
      ..setNavigationDelegate(NavigationDelegate(onProgress: (int progress) {
        debugPrint('WebView is loading (progress : $progress%)');
      }, onPageStarted: (String url) {
        if (!mounted) return;
        setState(() {
          isLoading = true;
        });
      }, onPageFinished: (String url) {
        if (!mounted) return;
        setState(() {
          isLoading = false;
        });
      }, onNavigationRequest: (NavigationRequest request) {
        if (request.url.startsWith('https://hlamer.ru')) {
          widget.callback(request.url);
          return NavigationDecision.prevent;
        }
        return NavigationDecision.navigate;
      }))
      ..loadRequest(Uri.parse(widget.url));

    if (_controller.platform is AndroidWebViewController) {
      (_controller.platform as AndroidWebViewController).setMediaPlaybackRequiresUserGesture(false);
    }
```

**Comment by deepak-agendaboa** [comment]
Same issue:

```
Fatal Exception: java.lang.IllegalStateException: Could not find identifier for WebView.
       at io.flutter.plugins.webviewflutter.WebViewClientFlutterApiImpl.onPageStarted(WebViewClientFlutterApiImpl.java:82)
       at io.flutter.plugins.webviewflutter.WebViewClientHostApiImpl$WebViewClientImpl.onPageStarted(WebViewClientHostApiImpl.java:49)
       at YB0.b(YB0.java:15)
       at Ea.handleMessage(Ea.java:538)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:233)
       at android.app.ActivityThread.main(ActivityThread.java:8010)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```

```
Fatal Exception: java.lang.IllegalStateException: Could not find identifier for WebView.
       at io.flutter.plugins.webviewflutter.WebChromeClientFlutterApiImpl.onProgressChanged(WebChromeClientFlutterApiImpl.java:43)
       at io.flutter.plugins.webviewflutter.WebChromeClientHostApiImpl$WebChromeClientImpl.onProgressChanged(WebChromeClientHostApiImpl.java:50)
       at Ea.handleMessage(Ea.java:278)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:233)
       at android.app.ActivityThread.main(ActivityThread.java:8010)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```

```
Fatal Exception: java.lang.IllegalStateException: Could not find identifier for JavaScriptChannel.
       at io.flutter.plugins.webviewflutter.JavaScriptChannelFlutterApiImpl.getIdentifierForJavaScriptChannel(JavaScriptChannelFlutterApiImpl.java:39)
       at io.flutter.plugins.webviewflutter.JavaScriptChannelFlutterApiImpl.postMessage(JavaScriptChannelFlutterApiImpl.java:33)
       at io.flutter.plugins.webviewflutter.JavaScriptChannel.lambda$postMessage$1(JavaScriptChannel.java:46)
       at io.flutter.plugins.webviewflutter.JavaScriptChannel.$r8$lambda$K7533Hqo0qg7ff56ym-eNCAOHa4(JavaScriptChannel.java)
       at io.flutter.plugins.webviewflutter.JavaScriptChannel$$InternalSyntheticLambda$4$bfb098af4bca021f4a8e2ad834eec76b466d9ae66cfba252fa3986162b62e327$0.run(JavaScriptChannel.java:4)
       at android.os.Handler.handleCallback(Handler.java:938)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:233)
       at android.app.ActivityThread.main(ActivityThread.java:8010)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:631)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:978)
```

**Comment by deepak-agendaboa** [comment]
Earlier we were getting the crash 

```
Fatal Exception: java.lang.AssertionError: Manager has already been closed.
```

But as this assert is now ignored according to the PR https://github.com/flutter/plugins/pull/6943 and in the release `webview_flutter_android: 3.1.3`

And now it started giving the above errors.

**Comment by DFelten** [comment]
Same issue here. Crashlytics shows me the following log:

```
Fatal Exception: java.lang.IllegalStateException: Could not find identifier for WebView.
       at io.flutter.plugins.webviewflutter.WebChromeClientFlutterApiImpl.onProgressChanged(WebChromeClientFlutterApiImpl.java:24)
       at io.flutter.plugins.webviewflutter.WebChromeClientHostApiImpl$WebChromeClientImpl.onProgressChanged(WebChromeClientHostApiImpl.java:9)
       at ua.handleMessage(ua.java:278)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:210)
       at android.os.Looper.loop(Looper.java:299)
       at android.app.ActivityThread.main(ActivityThread.java:8319)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1038)
```

<details>
  <summary>WebView widget</summary>

```dart
class TMXWebView extends StatefulWidget {
  const TMXWebView({
    required this.controller,
    this.uri,
    this.htmlString,
    this.sessionId,
    this.onNavigationRequest,
  });

  final WebViewController controller;
  final Uri? uri;
  final String? htmlString;
  final String? sessionId;
  final NavigationRequestCallback? onNavigationRequest;

  @override
  State<TMXWebView> createState() => _TMXWebViewState();
}

class _TMXWebViewState extends State<TMXWebView> {
  @override
  void initState() {
    super.initState();

    widget.controller
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..setNavigationDelegate(
        NavigationDelegate(
          onPageFinished: _onPageFinished,
          onNavigationRequest: widget.onNavigationRequest ?? _onNavigationRequest,
        ),
      );

    if (widget.uri != null) {
      widget.controller.loadRequest(
        widget.uri!,
        headers: widget.sessionId != null ? {'Authorization': 'Bearer ${widget.sessionId}'} : {},
      );
    } else if (widget.htmlString != null) {
      widget.controller.loadHtmlString(widget.htmlString!);
    }

    WidgetsBinding.instance.addPostFrameCallback(
      (_) => widget.controller.setBackgroundColor(tmxColors.background(context)),
    );
  }

  @override
  Widget build(BuildContext context) {
    return BlocBuilder<TMXWebViewBloc, TMXWebViewState>(
      builder: (context, state) => _buildWebView(context, state.pageFinished),
    );
  }

  Widget _buildWebView(BuildContext context, bool pageFinished) {
    return Stack(
      children: [
        WebViewWidget(
          controller: widget.controller,
        ),
        if (!pageFinished) const TMXLoadingIndicator(),
      ],
    );
  }

  Future<void> _onPageFinished(String url) async {
    context.read<TMXWebViewBloc>().add(
          const PageLoadFinished(),
        );
  }

  NavigationDecision _onNavigationRequest(NavigationRequest request) {
    /// Do some stuff

    return NavigationDecision.navigate;
  }
}
```

</details>

**Comment by vinaykharayat** [comment]
Same error occurred, one thing is that my default browser is Firefox, although I don't have chrome installed. Web-view works fine though.

**Comment by GeylanKalafMohe** [comment]
It is available as a new version on pub.dev?

**Comment by bparrishMines** [comment]
Yes, it has been published as version 3.4.1 for webview_flutter_android. Run `flutter pub upgrade` to get the latest version.
